src/agent/communicator/src/communicator.cpp

```diff
@@ -0,0 +1,90 @@
+#include "communicator.hpp"
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/stream.hpp>
+#include <boost/beast/version.hpp>
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+namespace net = boost::asio;
+using tcp = boost::asio::ip::tcp;
+using json = nlohmann::json;
+
+namespace communicator
+{
+
+    Communicator::Communicator()
+    {
+
+    }
+
+    int Communicator::SendAuthenticationRequest()
+    {
+        json bodyJson = {{communicator::uuidKey, communicator::kUUID},
+                         {communicator::passwordKey, communicator::kPASSWORD}};
+        http::response<http::dynamic_body> res =
+            this->sendHttpRequest(http::verb::post, "/authentication", "", bodyJson.dump());
```

Comment:
`this->` is not necessary, it is implied within the class scope.

Reply:
Done